tools/xabuild/SymbolicLink.cs

```diff
@@ -22,23 +21,23 @@ public static bool Create (string source, string target)
 						return result;
 					}
 				} else {
-					try {
-						var sourceInfo = new UnixFileInfo (source);
-						var fileInfo = new UnixFileInfo (target);
-						fileInfo.CreateSymbolicLink (source);
-					} catch (UnixIOException exc) {
-						if (exc.ErrorCode == Mono.Unix.Native.Errno.EEXIST) {
-							return true;
-						}
-						Console.Error.WriteLine ($"Unable to create symbolic link from `{source}` to `{target}`: {exc}");
-						return false;
-					}
+					return CreateUnixSymLink (source, target);
 				}
 			}
 
 			return true;
 		}
 
+		static bool CreateUnixSymLink (string source, string target)
+		{
+			int r = symlink (Path.GetFullPath (target), source);
+			if (r != 0) {
+				perror ($"`ln -s '{source}' '{target}'` failed");
```

Comment:
Would Console.Error.WriteLine be easier for CSharp folks to understand? I had to use google to learn what perror was.

Reply:
@jonathanpeppers: On the one hand, yes. On the other, **perror**(3) is the easiest way to get **strerror**(3) output, i.e. "why did it fail?".

Perhaps I should P/Invoke **strerror**(3) instead?

(Which might actually be *safer*, as it's entirely possible that the P/Invoke for ***perror**(3) itself* will change `errno`, which is what we want printed in the first place!)

Reply:
Then you would have to deal with calling `strerror` and passing it the value of `errno` which is not easily accessible from the managed world. `perror` is much easier as it does that behind the scenes.

Reply:
Oops, @jonpryor , I didn't see your comment (was typing while you posted), sorry about repeating what you've said :)

Reply:
...but **strerror**(3) isn't thread safe (argh!), and **strerror_r**(3) *is* thread safe, but (1) has multiple variants (GNU vs. XPG) and (2) uses `size_t`, which is always annoying to bind.

I'd prefer to stick with **perror**(3). At least it *could* work...